mcp/src/main/java/io/airlift/mcp/reflection/MethodInvoker.java

```diff
@@ -123,7 +121,7 @@ public Object invoke()
                                 case CallToolRequestParameter _ -> callToolRequest.orElseThrow(() -> new IllegalStateException("CallToolRequest is required"));
                                 case SourceResourceParameter _ -> sourceResource.orElseThrow(() -> new IllegalStateException("SourceResource is required"));
                                 case ReadResourceRequestParameter _ -> readResourceRequest.orElseThrow(() -> new IllegalStateException("ReadResourceRequest is required"));
-                                case IdentityParameter _ -> identityMapper.orElseThrow(() -> new IllegalStateException("IdentityMapper is required")).map(request);
+                                case IdentityParameter _ -> requireNonNull(request.getAttribute(MCP_IDENTITY_ATTRIBUTE));
```

Comment:
Move this into a helper method that throws `new IllegalStateException("IdentityMapper is required"))` etc.